### ISSUE TITLE
False positive error when subclassing `List` type

### ISSUE THREAD

**Issue by thepabloaguilar** [python/mypy#15040]
**Bug Report**

> This bug is experienced in [classes](https://github.com/dry-python/classes) lib

**To Reproduce**

```python
from typing import List


class ListOfStrMeta(type):
    def __instancecheck__(cls, other) -> bool:
        return (
            isinstance(other, list) and
            bool(other) and
            all(isinstance(list_item, str) for list_item in other)
        )


class ListOfStr(List[str], metaclass=ListOfStrMeta):
    ...

```

**Expected Behavior**

No errors should appear

**Actual Behavior**

Error from the code above:

```text
poetry run mypy ex.py
ex.py:13: error: Metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases  [misc]
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: 1.2.0
- Mypy command-line flags: N/A
- Mypy configuration options from `mypy.ini` (and other config files):
```cfg
[mypy]
# Plugins, includes custom:
plugins =
  classes.contrib.mypy.classes_plugin

disable_error_code = empty-body

allow_redefinition = false
check_untyped_defs = true
disallow_any_explicit = true
# disallow_any_generics = true
disallow_untyped_calls = true
ignore_errors = false
ignore_missing_imports = true
implicit_reexport = false
local_partial_types = true
strict_optional = true
strict_equality = true
no_implicit_optional = true
warn_no_return = true
warn_unused_ignores = true
warn_redundant_casts = true
warn_unused_configs = true
warn_unreachable = true
```
- Python version used: 3.10.10

cc @sobolevn

**Comment by ikonst** [comment]
This is probably due to #6042:
https://github.com/python/mypy/blob/0f09be472cad7aa06ca4af9dd73aca6b87450d84/mypy/semanal.py#L2363-L2372
Can take a look @ilevkivskyi?

Since `list` derives from a handful of protocols (e.g. `Sequence`), its metaclass is `ABCMeta`.

Changing
```diff
-class ListOfStrMeta(typing):
+class ListOfStrMeta(ABCMeta):
```
[eliminates](https://gist.github.com/mypy-play/8f137fdc200270c787c542511aeb9a63) the error.

**Comment by sobolevn** [comment]
Yes, this is due to the fact that `typeshed` has `Sequence` in the base classes of `list`
While runtime does not have it.

**Comment by hauntsaninja** [comment]
https://github.com/python/mypy/pull/17713 makes the error here go away
